### ISSUE TITLE
Unexpected info: new_offline_messages

### ISSUE THREAD

**Issue by Beisenbek** [esl/MongooseIM#1754]
MongooseIM version: 2.1.1
Installed from: source
Erlang/OTP version: 18

Time by time we getting next errors:
`2018-03-06 05:19:45.996 [error] <0.1641.0>@ejabberd_c2s:handle_incoming_message:1275 Unexpected info: new_offline_messages
2018-03-06 05:19:47.407 [error] <0.1641.0>@ejabberd_c2s:handle_incoming_message:1275 Unexpected info: new_offline_messages`

What does it mean *unexpected*? Why these messages are not handled by push module?

**Comment by fenek** [comment]
Hi @Beisenbek 

Such error means some strange race condition. `new_offline_messages` is a message sent to user session by offline storage process. It is handled by the session only when it is fully established.

Do you know a consistent way to reproduce the error?

In general, `mod_offline` usage is discouraged as `mod_mam` is the preferred method of offline messages synchronisation between devices.

**Comment by Beisenbek** [comment]
Sure.

Precondition:
1. UserA and UserB installed mobile android clients with Smack. 
2. `   {mod_mam, []},
   {mod_mam_riak_timed_arch_yz, [pm,muc]},`
3. `{mod_offline, [{access_max_user_messages, max_user_offline_messages}]},`
4. `  {mod_pubsub, [  
               {plugins, [<<"push">>]}
  ]}`

Steps for reproducing:
1. UserA is online.
2. UserB is offline (client app in background mode, not closed, but sent to server "unavailable" presence).
3. UserA sent to UserB message.
4. UserB got push notification.
5. In MongooseIM: log/ejabberd.log shows: `2018-03-15 03:58:57.230 [error] <0.20726.1>@ejabberd_c2s:handle_incoming_message:1311 Unexpected info: new_offline_messages`

**Comment by fenek** [comment]
Does your app close TCP connection when going to background?

Also, can you please:
1. Start MongooseIM in interactive mode (`mongooseimctl live`)
2. Reduce log verbosity (`ejabberd_loglevel:set(1).`)
3. Enable trace
```
dbg:tracer().
dbg:p(all, c).
dbg:tp(ejabberd_c2s, handle_info, x).
```
4. Copy & paste the console output after you reproduce the issue?

**Comment by Beisenbek** [comment]
No, app doesn't close connection.

Please find in attached [file](https://github.com/esl/MongooseIM/files/1818007/logs.txt) output from console.

**Comment by Beisenbek** [comment]
This issue is reproducible with UserB in flight mode (i.e without any internet connection)

**Comment by fenek** [comment]
Thanks, these logs helped a lot. :) Indeed the handling of `new_offfline_messages` event should be improved in `ejabberd_c2s` module, as the condition you're reproducing is definitely not an error.

As far as I understand, except for misleading error log message, everything works as expected?

**Comment by Beisenbek** [comment]
Many thanks for information.

Everything works as expected except that we have message losses about 0.1%, so I'm trying to handle all errors in logs. I see that this misleading error not related with losses.

Will `new_offfline_messages event`  improved for avoiding misleading error log message in nearest commits?

**Comment by fenek** [comment]
Depends on how you define "nearest". :) I can say at least that it is fairly probable that it will be fixed until 3.0 release (mid-April).

**Comment by ludwikbukowski** [comment]
the same when combined stream management + offline messages
